src/sdk/account/utils/Types.ts

```diff
@@ -94,6 +94,8 @@ export type TypeDefinition = {
   [key: string]: TypeField[]
 }
 
+// Review: NonceOptions already has nonceKey and validationMode
+// Why is this key needed?
```

Comment:
This key comes from viem. viem userse expect to be able to set the nonce of txs using this param because `SmartAccount.getNonce()` method has the type: 

`(parameters?: { key?: bigint | undefined; } | undefined) => Promise<bigint>'`

Reply:
ok so it's nonceKey only.
I am confused because we also have NonceOptions which repeats these two things validationMode and nonceKey = key.

Reply:
also, how would I pass it from sendTransaction / sendUserOperation? 
PR mentions I have't added test cases yet.

Reply:
prepareUserOperation has a nonce field that you can set (alongside the calls).

So I presume you can set a new nonce like so:

nexusClient.prepareUserOperation({
  calls: [],
  nonce: await nexusClient.account.getNonce(...some custom params)
})

I'm guessing a bit. Could look under the hood to see what's going on. I notice that the nexusAccount.getNonce() gets called under the hood anyway as userOps are sent. I guess this happens if a nonce isn't explicitly set in the userOp? Requires investigation.

Reply:
Yea exactly, this is what happens, you can provide a custom nonce which should be, in our case, the 2d nonce, if you provide it then viem will not call .getNonce() on their side anymore

Reply:
I'd like to pass nonceKey in getNonce function. or override nonce having nonce calculated already using my nonceKey.

Reply:
I noticed prepareUserOperation does, but how do I make it propagate from sendTransaction and sendUserOp ?

Reply:
Does sendTransaction not ultimately call prepareUserOperation under the hood? I'm guessing that this might work too?

```
nexusClient. sendTransaction({
calls: [],
nonce: await nexusClient.account.getNonce(...some custom params)
})
```